source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+
+
+File Format
+===========
+
+=============  ===========================================
+**Format**     JSON (with comments)
+
+**Extension**  ``.sublime-menu``
+
+**Name**       One out of the list of available menus.
+               See :ref:`menu-types` for the complete list
+               and what menu they represent.
+
+**Location**   Any
+
+**Content**    A list of "menu item" objects.
+=============  ===========================================
+
+Example
+*******
+
+The following is an excerpt
+from the default :file:`Main.sublime-menu` file.
+
+.. code-block:: json
+
+   [
+       {
+           "caption": "Edit",
+           "mnemonic": "E",
+           "id": "edit",
+           "children":
+           [
+               { "command": "undo", "mnemonic": "U" },
+               { "command": "redo_or_repeat", "mnemonic": "R" },
+               {
+                   "caption": "Undo Selection",
+                   "children":
+                   [
+                       { "command": "soft_undo" },
+                       { "command": "soft_redo" }
+                   ]
+               },
+               { "caption": "-", "id": "clipboard" },
+               { "command": "copy", "mnemonic": "C" },
+               { "command": "cut", "mnemonic": "t" },
+               { "command": "paste", "mnemonic": "P" },
+               { "command": "paste_and_indent", "mnemonic": "I" },
+               { "command": "paste_from_history", "caption": "Paste from History" }
+           ]
+       }
+   ]
+
+
+Images
+******
+
+.. figure:: images/context_menu_default.png
+   :class: shadowed
+   :figclass: float-left
+   :figwidth: 50%
+
+   The default context menu in the editing area.
+
+.. figure:: images/context_menu_modified.png
+   :class: shadowed
+   :figclass: float-left
+   :figwidth: 50%
+
+   A modified context menu in the editing area.
+
+
+.. _menu-types:
+
+Available Menus
+===============
+
+The file name
+of a ``.sublime-menu`` file
+specifies the menu in the application
+to be modified.
+
+The following menus are available:
+
+========================   ===============
+     File/Menu Name          Description
+========================   ===============
+**Main**                   Main menu
+
+**Context**                Context menu in the editing area
+
+**Find in Files**          Appears when clicking the "…" button
+                           in the :doc:`Find in Files
+                           </search_and_replace/search_and_replace_files>` panel
+
+**Side Bar**               Context menu for each node in the sidebar
+
+**Side Bar Mount Point**   Additional context menu items
+                           for the top-level nodes in the sidebar
+
+**Tab Context**            Context menu of the tab bar
+
+**Widget Context**         Context menu of input fields
+                           in all kinds of widgets,
+                           including Command Palette, Goto Anything,
+                           the find panels

```

Comment:
"find" can be both a verb and a substantive, just like "search". I will just capitalize "Find", like we do with Goto Anything and Command Palette. That should make it clear.

And well, Sublime Text does indeed call it "Incremental Find".